apps/meteor/server/models/raw/LivechatVisitors.ts

```diff
@@ -365,6 +366,32 @@ export class LivechatVisitorsRaw extends BaseRaw<ILivechatVisitor> implements IL
 			},
 		);
 	}
+
+	isVisitorActiveOnPeriod(visitorId: string, period: string): Promise<boolean> {
+		const query = {
+			_id: visitorId,
+			activity: period,
+		};
+
+		return this.findOne(query, { projection: { _id: 1 } }).then(Boolean);
+	}
+
+	markVisitorActiveForPeriod(visitorId: string, period: string): Promise<UpdateResult> {
+		const query = {
+			_id: visitorId,
+		};
+
+		const update = {
+			$push: {
+				activity: {
+					$each: [period],
+					$slice: -12,
```

Comment:
Any specific reason for us to keep only the most 12 recent ones? 🤔

Reply:
No actual reason, just thought 1Y of data would be enough for most cases. We can increase it, what i don't want is keeping it unbounded.